File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/attribute_provider.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

public class attribute_provider : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal attribute_provider(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(attribute_provider obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~attribute_provider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_attribute_provider(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public attribute_provider() : this(gnsdk_csharp_marshalPINVOKE.new_attribute_provider(), true) {
  }

  public string get_data(uint pos) {
    string ret = gnsdk_csharp_marshalPINVOKE.attribute_provider_get_data(swigCPtr, pos);
    return ret;
  }

  public uint count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.attribute_provider_count(swigCPtr);
    return ret;
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/attribute_provider.java

package com.gracenote.gnsdk;

public class attribute_provider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected attribute_provider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(attribute_provider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_attribute_provider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public attribute_provider() {
    this(gnsdk_javaJNI.new_attribute_provider(), true);
  }

  public String getData(long pos) {
    return gnsdk_javaJNI.attribute_provider_getData(swigCPtr, this, pos);
  }

  public long count() {
    return gnsdk_javaJNI.attribute_provider_count(swigCPtr, this);
  }

}
